src/cascadia/TerminalControl/TermControl.cpp

```diff
@@ -703,13 +706,31 @@ namespace winrt::Microsoft::Terminal::TerminalControl::implementation
                 const auto cursorPosition = point.Position();
                 const auto terminalPosition = _GetTerminalPosition(cursorPosition);
 
-                // save location before rendering
-                _terminal->SetSelectionAnchor(terminalPosition);
-
                 // handle ALT key
                 _terminal->SetBoxSelection(altEnabled);
 
-                _renderer->TriggerSelection();
+                if (_IsTripleClick(cursorPosition, point.Timestamp()))
+                {
+                    _terminal->TripleClickSelection(terminalPosition);
+                    _renderer->TriggerSelection();
+                    _doubleClickOccurred = true;

```

Comment:
`_IsTripleClick` checks if `_doubleClickOccurred` is true/false before proceeding.

Basically, it is impossible to get here to set `_doubleClickOccurred` to true unless it was already true to get past that gate.

Reply:
Yeah. So that line is actually redundant. Good catch.